modules/svar_master/1.0/src/combine_annotated_bedpe.R

```diff
@@ -0,0 +1,111 @@
+suppressWarnings({
+  suppressPackageStartupMessages({
+    library(tidyverse)
+  })
+  
+  final_cols <- c(
+    "CHROM_A", 
+    "START_A", 
+    "END_A",
+    "CHROM_B", 
+    "START_B", 
+    "END_B",
+    "manta_name", 
+    "SCORE", 
+    "STRAND_A",
+    "STRAND_B",
+    "tumour_sample_id",
+    "normal_sample_id",
+    "VAF",
+    "DP",
+    "ANNOTATION_A", 
+    "DIST_TO_ANNOTATION_A", 
+    "ANNOTATION_B", 
+    "DIST_TO_ANNOTATION_B",
+    "gridss_name",
+    "delly_name"
+  )
+  
+  # Load and tidy combined GRIDSS and Manta bedpe annotations
+  combined_a <- read_tsv(snakemake@input[["a_combined"]], 
+                         col_types = cols(
+                           VAF = col_double()
+                         ),
+                         col_names <- c(
+                           "CHROM_A", 
+                           "START_A", 
+                           "END_A", 
+                           "manta_name",
+                           "SCORE",
+                           "STRAND_A",
+                           "STRAND_B",
+                           "tumour_sample_id",
+                           "normal_sample_id",
+                           "VAF",
+                           "DP",
+                           "gridss_name", 
+                           "CHROM_ANN_A", 
+                           "START_ANN_A", 
+                           "END_ANN_A", 
+                           "ANNOTATION_A", 
+                           "DIST_TO_ANNOTATION_A"
+                         ))
+  
+  combined_b <- read_tsv(snakemake@input[["b_combined"]], 
+                         col_types = cols(
+                           VAF = col_double()
+                         ), 
+                         col_names <- c(
+                           "CHROM_B", 
+                           "START_B", 
+                           "END_B", 
+                           "manta_name",
+                           "SCORE",
+                           "STRAND_A",
+                           "STRAND_B",
+                           "tumour_sample_id",
+                           "normal_sample_id",
+                           "VAF",
+                           "DP",
+                           'gridss_name', 
+                           "CHROM_ANN_B", 
+                           "START_ANN_B", 
+                           "END_ANN_B", 
+                           "ANNOTATION_B", 
+                           "DIST_TO_ANNOTATION_B"
+                         ))
+  
+  
+  
+  if(nrow(combined_a) > 0){
```

Comment:
should there be a warning/error message if there are 0 rows in the input bed? Just in case the input is empty, I imagine the 0 rows would give a not straightforward error message. But this can also be kept in mind for future updates and next versions, I think it does not necessarily apply for the samples we've seen so far :smile:

Reply:
Yep, this if statement was added because I did have 0-row files when I developed this script for the capture data. This allows the pipeline to run to completion even if the bed file is empty.